pkg/sql/catalog/internal/catkv/catalog_reader_cached.go

```diff
@@ -220,6 +221,13 @@ func (c *cachedCatalogReader) ScanAll(ctx context.Context, txn *kv.Txn) (nstree.
 	return read, nil
 }
 
+// ScanDescriptorsInSpans is part of the CatalogReader interface.
+func (c *cachedCatalogReader) ScanDescriptorsInSpans(
```

Comment:
Unsure of a good caching strategy and when cache invalidation occurs, looking for input if any